expire/expire_test.go

```diff
@@ -0,0 +1,147 @@
+package expire
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/contribsys/faktory/cli"
+	"github.com/contribsys/faktory/client"
+	"github.com/contribsys/faktory/server"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExpiresAt(t *testing.T) {
+	withServer(func(s *server.Server, cl *client.Client) {
+		// Push a job that has already expired
+		j1 := client.NewJob("JobOne", 1)
+		j1.SetExpiresAt(time.Now().Add(time.Minute * -1))
+		err := cl.Push(j1)
+		assert.Nil(t, err)
+
+		// Push a job that will expire
+		j2 := client.NewJob("JobOne", 1)
+		j2.SetExpiresAt(time.Now().Add(time.Minute * 1))
+		err = cl.Push(j2)
+		assert.Nil(t, err)
+
+		// The fetched job should be j2 because j1 is expired
+		job, err := cl.Fetch("default")
+		assert.Nil(t, err)
+		assert.Equal(t, j2.Jid, job.Jid)
+
+		// Verify that the queue is empty
+		job, err = cl.Fetch("default")
+		assert.Nil(t, err)
+		assert.Nil(t, job)
+	})
+}
+
+func TestExpiresIn(t *testing.T) {
+	withServer(func(s *server.Server, cl *client.Client) {
+		// Push a job that has already expired
+		j1 := client.NewJob("JobOne", 1)
+		j1.SetExpiresIn(time.Minute * -1)
+		err := cl.Push(j1)
+		assert.Nil(t, err)
+
+		// Push a job that will expire
+		j2 := client.NewJob("JobOne", 1)
+		j2.SetExpiresIn(time.Minute * 1)
+		err = cl.Push(j2)
+		assert.Nil(t, err)
+
+		// The fetched job should be j2 because j1 is expired
+		job, err := cl.Fetch("default")
+		assert.Nil(t, err)
+		assert.Equal(t, j2.Jid, job.Jid)
+
+		// Verify that the queue is empty
+		job, err = cl.Fetch("default")
+		assert.Nil(t, err)
+		assert.Nil(t, job)
+	})
+}
+
+// The `expires_in` value is converted to a timestamp and saved as `expires_at`
+func TestRemovesExpiresIn(t *testing.T) {
+	withServer(func(s *server.Server, cl *client.Client) {
+		// Push a job that has already expired
+		j1 := client.NewJob("JobOne", 1)
+		j1.SetCustom("expires_in", (time.Duration(1) * time.Minute).String())
```

Comment:
You might also want to test that this works when you `j1.SetExpiresIn` (which is how consumers will do it) rather than `j1.SetCustom` directly.

Reply:
That's covered in the previous test